lib/msal-node/src/client/ManagedIdentitySources/ServiceFabric.ts

```diff
@@ -3,17 +3,155 @@
  * Licensed under the MIT License.
  */
 
+import { INetworkModule, Logger } from "@azure/msal-common";
 import { ManagedIdentityId } from "../../config/ManagedIdentityId";
 import { ManagedIdentityRequestParameters } from "../../config/ManagedIdentityRequestParameters";
-import { BaseManagedIdentitySource } from "./BaseManagedIdentitySource";
+import {
+    BaseManagedIdentitySource,
+    getValidatedEnvVariableUrlString,
+} from "./BaseManagedIdentitySource";
+import { NodeStorage } from "../../cache/NodeStorage";
+import { CryptoProvider } from "../../crypto/CryptoProvider";
+import {
+    API_VERSION_QUERY_PARAMETER_NAME,
+    HttpMethod,
+    ManagedIdentityEnvironmentVariableNames,
+    ManagedIdentityIdType,
+    ManagedIdentitySourceNames,
+    RESOURCE_QUERY_PARAMETER_NAME,
+    SERVICE_FABRIC_SECRET_HEADER_NAME,
+} from "../../utils/Constants";
 
+// MSI Constants. Docs for MSI are available here https://docs.microsoft.com/azure/app-service/overview-managed-identity
+const SERVICE_FABRIC_MSI_API_VERSION: string = "2019-07-01-preview";
+
+/**
+ * Original source of code: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/identity/Azure.Identity/src/ServiceFabricManagedIdentitySource.cs
+ */
 export class ServiceFabric extends BaseManagedIdentitySource {
+    private identityEndpoint: string;
+    private identityHeader: string;
+
+    constructor(
+        logger: Logger,
+        nodeStorage: NodeStorage,
+        networkClient: INetworkModule,
+        cryptoProvider: CryptoProvider,
+        identityEndpoint: string,
+        identityHeader: string,
+        systemAssigned: boolean
+    ) {
+        super(logger, nodeStorage, networkClient, cryptoProvider);
+
+        this.identityEndpoint = identityEndpoint;
+        this.identityHeader = identityHeader;
+
+        if (systemAssigned) {
+            logger.warning(
+                `[Managed Identity] ${ManagedIdentitySourceNames.SERVICE_FABRIC} user assigned managed identity ClientId or ResourceId is not configurable at runtime.`
```

Comment:
I don't understand this warning. What does it mean?

Reply:
See out teams chat for more information and please let me know your thoughts.